File: spring-best/src/main/java/com/best/springbest/model/Voucher.java
package com.best.springbest.model;


import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.format.annotation.DateTimeFormat;

import javax.persistence.*;
import java.util.Date;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "vouchers")
public class Voucher {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String code;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "shop_id",nullable = false)
    private Shop shop;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "product_id",nullable = false)
    private Product product;

    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private Date date_start;

    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private Date date_end;

    private int discount_percent;

    private String description;
}

File: BEST-App-main/lib/model/voucher.dart
class Voucher {
  final int id;
  final String code;
  final int shopId;
  final int productId;
  final String dateStart;
  final String dateEnd;
  final int discountPercent;
  final String description;

  Voucher(this.id, this.code,this.shopId, this.productId ,this.dateStart, this.dateEnd,
      this.discountPercent, this.description);

  static Map<String, dynamic> toJson(Voucher voucher) {
    return {
      'id': voucher.id,
      'code': voucher.code,
      'shop_id': voucher.shopId,
      'product_id': voucher.productId,
      'date_start': voucher.dateStart,
      'date_end': voucher.dateEnd,
      'discount_percent': voucher.discountPercent,
      'description': voucher.description
    };
  }
}
